File: backend/bin/main/com/smarttour/model/Place.kt
package com.smarttour.model

import jakarta.persistence.*
import java.time.LocalDateTime
import java.util.UUID

@Entity
@Table(name = "places")
data class Place(
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    val id: UUID? = null,
    
    @Column(nullable = false)
    val name: String,
    
    @Column(length = 2000)
    val description: String? = null,
    
    @Column(nullable = false)
    val latitude: Double,
    
    @Column(nullable = false)
    val longitude: Double,
    
    @Enumerated(EnumType.STRING)
    val category: PlaceCategory? = PlaceCategory.OTHER,
    
    val rating: Double? = null,
    
    val imageUrl: String? = null,
    
    val address: String? = null,
    
    val googlePlaceId: String? = null,
    
    @Column(updatable = false)
    val createdAt: LocalDateTime = LocalDateTime.now()
) {
    constructor() : this(
        id = null,
        name = "",
        description = null,
        latitude = 0.0,
        longitude = 0.0,
        category = PlaceCategory.OTHER,
        rating = null,
        imageUrl = null,
        address = null,
        googlePlaceId = null,
        createdAt = LocalDateTime.now()
    )
}

enum class PlaceCategory {
    RESTAURANT, CAFE, MUSEUM, PARK, HISTORIC, SHOPPING, 
    ENTERTAINMENT, NATURE, LANDMARK, RELIGIOUS, OTHER
}

File: frontend/lib/models/place.dart
class Place {
  final String? id;
  final String name;
  final String? description;
  final double latitude;
  final double longitude;
  final String? category;
  final double? rating;
  final String? imageUrl;
  final String? address;
  final double? distanceMeters;

  Place({
    this.id,
    required this.name,
    this.description,
    required this.latitude,
    required this.longitude,
    this.category,
    this.rating,
    this.imageUrl,
    this.address,
    this.distanceMeters,
  });

  factory Place.fromJson(Map<String, dynamic> json) {
    return Place(
      id: json['id'],
      name: json['name'] ?? 'Unknown',
      description: json['description'],
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
      category: json['category'],
      rating: json['rating'] != null ? (json['rating'] as num).toDouble() : null,
      imageUrl: json['imageUrl'],
      address: json['address'],
      distanceMeters: json['distanceMeters'] != null ? (json['distanceMeters'] as num).toDouble() : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'latitude': latitude,
      'longitude': longitude,
      'category': category,
      'rating': rating,
      'imageUrl': imageUrl,
      'address': address,
    };
  }
}

class NearbyPlacesResponse {
  final List<Place> places;
  final double userLatitude;
  final double userLongitude;
  final int totalFound;

  NearbyPlacesResponse({
    required this.places,
    required this.userLatitude,
    required this.userLongitude,
    required this.totalFound,
  });

  factory NearbyPlacesResponse.fromJson(Map<String, dynamic> json) {
    return NearbyPlacesResponse(
      places: (json['places'] as List).map((e) => Place.fromJson(e)).toList(),
      userLatitude: (json['userLatitude'] as num).toDouble(),
      userLongitude: (json['userLongitude'] as num).toDouble(),
      totalFound: json['totalFound'] ?? 0,
    );
  }
}
